### ISSUE TITLE
tedge-mapper-c8y inventory not working

### ISSUE THREAD

**Issue by scharri** [thin-edge/thin-edge.io#1496]
**Describe the bug**
No fragments defined in the inventory.json file get send to the cloud.

**To Reproduce**
1. Create inventory.json in base_config_dir (/etc/tedge/device/inventory.json)
2. Add content (I used the same as in the documentation: [How to Fragments](https://thin-edge.github.io/thin-edge.io/html/howto-guides/022_c8y_fragments.html)
3. sudo systemctl restart tedge-mapper-c8y.service

**Expected behavior**
Defined properties should be sent to the cloud.

**Environment (please complete the following information):**
 - OS: Debian GNU/Linux 11 (bullseye)
 - Hardware: Raspberry Pi 4 Model B Rev 1.4 (Revision: d03114)
 - System-Architecture: Linux raspberrypi 5.15.73-v8+ #1593 SMP PREEMPT Wed Oct 12 14:24:59 BST 2022 aarch64 GNU/Linux
 - thin-edge.io version: 0.7.5
 - mosquitto version: 2.0.11

**Additional context**
Output of subscription of c8y/# after tedge-mapper-c8y restart:
- c8y/s/uat (null)
- c8y/s/dat 71,<received JWT token>
- c8y/s/us 114,c8y_SoftwareUpdate,c8y_Restart,c8y_DownloadConfigFile,c8y_LogfileRequest,c8y_UploadConfigFile
- c8y/inventory/managedObjects/update/pi-10000000079b101b {"type":"TEST-RPi"}
- c8y/s/us 500
- c8y/s/uat (null)
- c8y/s/dat 71,<received JWT token>

**Comment by PradeepKiruvale** [comment]
Thanks for raising the issue. Yes, there is definitely an issue, we will fix this soon.

**Comment by PradeepKiruvale** [comment]
Found the issue and we are publishing on the wrong `topic`. So, the `fragment` message is getting lost and not reaching the c8y cloud. We will fix it soon.

**Comment by rina23q** [comment]
This bug was introduced by #1425, and confirmed that it affects only 0.7.5.

**Comment by scharri** [comment]
Thank you for your quick reply and confirmation of the bug. 
Unfortunately, I have not yet dealt with Rust, otherwise I would have already looked into the code myself.

**Comment by PradeepKiruvale** [comment]
@scharri Fixed and merged it already.

**Comment by PradeepKiruvale** [comment]
Resolved #1498

**Comment by PradeepKiruvale** [comment]
How to test:

Install the thin-edge from here https://github.com/thin-edge/thin-edge.io/actions/runs/3278493185

Create an inventory.json file with below content in `/etc/tedge/device`

```
{
  "c8y_RequiredAvailability": {
      "responseInterval": 5
  },
  "c8y_Firmware": {
      "name": "raspberrypi-bootloader",
      "version": "1.20140107-1",
      "url": "31aab9856861b1a587e2094690c2f6e272712cb1"
  },
  "c8y_Hardware": {
      "model": "BCM2708",
      "revision": "000e",
      "serialNumber": "00000000e2f5ad4d"
  }
}
```

Connect to c8y cloud

`sudo tedge connect c8y`

Verify if the fragments are uploaded onto the c8y cloud under `device->info section`

**Comment by anonymosh** [comment]
I seem to have hit the same bug in 0.7.7.

**Comment by rina23q** [comment]
> I seem to have hit the same bug in 0.7.7.

The fix will be delivered in 0.8.0.

**Comment by rina23q** [comment]
We delivered the fix in 0.8.0. @scharri Can we close this issue?

**Comment by scharri** [comment]
@rina23q you can close this issue. Let me know if i should close this issue.
